apps/web/src/services/dashboards/widgets/asset-widgets/severity-status-by-service/SeverityStatusByServiceWidget.vue

```diff
@@ -137,19 +142,34 @@ const fetchData = async (): Promise<Data[]> => {
 
 
 /* Util */
-const refineData = (data): Data[] => {
-    const { results } = data;
-    return results.map((result) => {
-        const minSeverityPriority: number = min(result.severity.map((s) => SEVERITY_STATUS_MAP[s].priority)) ?? 0; // ex. 1
-        const severity = SEVERITY_PRIORITY_MAP[minSeverityPriority]; // ex. 'CRITICAL'
-        const complianceStatus = result.status.includes('FAIL') ? 'FAIL' : 'PASS';
-        return {
-            service: result.service,
+const refineData = (data?: Data[]): RefinedData[] => {
+    if (!data?.length) return [];
+    const serviceSet = new Set();
+    data.forEach((result) => serviceSet.add(result.service));
+    const refinedData: RefinedData[] = [];
+    serviceSet.forEach((service) => {
+        const targetServiceDataList = data.filter((result) => result.service === service);
+        const statusList = flattenDeep(targetServiceDataList.map((d) => d.status));
+        const status = statusList.includes('FAIL') ? 'FAIL' : 'PASS';
+        let severity: Severity = 'PASS';
```

Comment:
nitpick: please separate logics to another function for the loosely related concern. e.g. `getStatus()`